src/Uno.UI.Runtime.Skia.Win32/Win32EventLoop.cs

```diff
@@ -130,13 +130,16 @@ public static void RunOnce(TimeSpan? timeout = null)
 				var cts = new CancellationTokenSource();
 				_ = Task.Run(async () =>
 				{
-					await Task.Delay(timeout.Value, cts.Token);
-
-					if (!cts.IsCancellationRequested)
+					try
 					{
+						await Task.Delay(timeout.Value, cts.Token);
```

Comment:
This is not enough... we're still getting cancelled exceptions. We're in a thread, we can use Thread.Sleep correctly here, though we'll need to handle cancellation properly from the token.